Add Header component

Arlabunakty opened the pull request.

Arlabunakty committed:
Add Header with user statistics

Arlabunakty committed:
Merge branch 'header' of github.com:Arlabunakty/git-cat into header

Arlabunakty committed:
Fix using assets for GitHub pages

inna-i commented on src/components/Header.js:
```diff
@@ -0,0 +1,40 @@
+import octocatLogo from "./../assets/octocat.png";
+
+const Header = ({ followers, forks, received_events }) => (
+  <>
```
Даний обгортка не потрібна, так як <div className="header-block"> обгортає весь контент

inna-i commented on src/components/Header.js:
```diff
@@ -0,0 +1,40 @@
+      <div className="block">
+        <table className="text">
+          <tbody>
+            <tr>
```
так як tr повторюються, то тут теж можна створити масив із конфігами і рендерити використовуючи
` rows.map(r => <tr>....</tr>)`

inna-i commented on src/components/Header.css:
```diff
@@ -0,0 +1,46 @@
+  border-top-right-radius: 1em;
+  border-bottom-right-radius: 1em;
+  border-top-left-radius: 1em;
+  border-bottom-left-radius: 1em;
```
Так як значення однакові, можна об'єднати 4 в 1: 
```suggestion
  border-radius: 1em;
```

inna-i commented on src/components/Header.css:
```diff
@@ -0,0 +1,46 @@
+.header-block {
```
можна використати простіший клас - .header

inna-i commented on src/components/Header.css:
```diff
@@ -0,0 +1,46 @@
+.block {
+  margin: 0 0.9em;
+  box-sizing: border-box;
+  display: inline-block;
```
Клас, який використовується дещо суперечить даному стилю :)

inna-i replied:
для позиціонування можна використити flexbox

Arlabunakty committed:
Fix PR review comments:
1. fix styling
2. simplify constructs
3. polish css file and namings
4. refactor tr rendering with configuration array
